velox/exec/MergeJoin.cpp

```diff
@@ -261,13 +270,34 @@ void MergeJoin::addOutputRowForLeftJoin(
   ++outputSize_;
 }
 
+void MergeJoin::flattenRightProjections() {
+  auto& children = output_->children();
+
+  for (const auto& projection : rightProjections_) {
+    auto& currentVector = children[projection.outputChannel];
+    auto newFlat = BaseVector::create(
+        currentVector->type(), outputBatchSize_, operatorCtx_->pool());
```

Comment:
outputBatchSize_ -> outputSize_ ? 

Does this work now because both values happen to be the same?

Reply:
The point here is that we flatten, but we may continue adding results. So we allocate the buffer to fit a full `outputSize_`